readme.md

```diff
@@ -1737,6 +1738,66 @@ export default MDXContent
 *   ± same as `main` branch of `@mdx-js/mdx`
 *   Fix JSX tags to prevent `<p><h1 /></p>`
 
+## Architecture
+
+To understand what this project does, it’s very important to first understand
+what unified does: please read through the
+[`unifiedjs/unified`](https://github.com/unifiedjs/unified) readme (the part
+until you hit the API section is required reading).
+
+**xdm** is a unified pipeline — wrapped so that most folks don’t need to know
+about unified.
+
+The *input* is MDX (serialized markdown with embedded JSX, ESM, and
+expressions).
+The markdown is parsed with [`micromark`][micromark] and the embedded JS with
+one of its extensions
+[`micromark-extension-mdxjs`](https://github.com/micromark/micromark-extension-mdxjs)
+(which in turn uses [acorn][]).
+Then [`mdast-util-from-markdown`](https://github.com/syntax-tree/mdast-util-from-markdown)
+and its extension
+[`mdast-util-mdx`](https://github.com/syntax-tree/mdast-util-mdx) are used to
+turn the results from the parser into a syntax tree:
+[mdast](https://github.com/syntax-tree/mdast).
+
+Markdown is closest to the source format.
+This is where remark plugins come in.
```

Comment:
it may be good to link to https://github.com/remarkjs/remark/blob/main/doc/plugins.md#plugins